src/cuda/device_util.cuh

```diff
@@ -171,7 +173,7 @@ inline float abs(cuFloatComplex x)
         b = fabsf(b);
         w = max(a, b);
         z = min(a, b);
-        if (z == 0) {
+        if (z == 0 || isinf( w )) {
```

Comment:
`z == 0` handles the case when `w == z == 0` that would cause `z / w = 0.0 / 0.0 = NaN` below, and otherwise (for `w != 0`) is an optimization.

`isinf( w )` handles case `z == w == inf` that would cause `z / w = inf / inf = NaN` below, and otherwise (for `z != inf`) is an optimization. This reflects change in `lapy2` in reference-LAPACK/lapack@906adae2c.